### ISSUE TITLE
JP2 as encoding standard id still throws and error

### ISSUE THREAD

**Issue by jordanpadams** [NASA-PDS-Incubator/validate#1210]
### Checked for duplicates

Yes - I've already checked

### 🐛 Describe the bug

When I tried to validate against a product with JP2 encoding standard id, I still get an internal error

### 🕵️ Expected behavior

I expected to not get an internal error

### 📜 To Reproduce

```
$ validate --disable-context-mismatch-warnings --skip-content-validation model-lddtool/src/test/resources/data/update_version/github905/test_case_Valid_20241024.xml

...

  FAIL: file:/Users/jpadams/proj/pds/pdsen/workspace/pds4-information-model/model-lddtool/src/test/resources/data/update_version/github905b/test_case_Valid_20241024.xml
      ERROR  [error.validation.internal_error]   Could not process the encoding type: encoding parameter 'JP2' is not known to this version of validate.
        1 product validation(s) completed
```

### 🩺 Test Data / Additional context

[github905b.zip](https://github.com/user-attachments/files/19674378/github905b.zip)

Can be tested with: https://github.com/NASA-PDS/validate/issues/1104

**Comment by jordanpadams** [comment]
This is blocking an IM change https://github.com/NASA-PDS/pds4-information-model/issues/904

**Comment by al-niessner** [comment]
@jordanpadams 

Next up in the batting cage.

https://github.com/NASA-PDS/validate/blob/1cae7ad5ff567cdd565a06a0f73374ae94d0be76/src/main/java/gov/nasa/pds/tools/util/EncodingMimeMapping.java#L10

says it should work. Are absolutely sure using latest version?

**Comment by al-niessner** [comment]
@jordanpadams 

Oh, I see the confusion. #1104 adds 'jp2' as a file extension NOT an encoding type. That needs to be J2C. Did we want to do something else and #1104 had it wrong?

with `<encoding_standard_id>J2C</encoding_standard_id>`:
```
Same as: validate --skip-context-validation --skip-content-validation -t /home/niessner/Projects/PDS/validate/src/test/resources/github1210/test_case_Valid_20241024.xml

PDS Validate Tool Report

Configuration:
   Version     3.7.0-SNAPSHOT
   Date        2025-04-09T22:20:30Z

Parameters:
   Targets                      [file:/home/niessner/Projects/PDS/validate/src/test/resources/github1210/test_case_Valid_20241024.xml]
   Severity Level               WARNING
   Recurse Directories          true
   File Filters Used            [*.xml, *.XML]
   Data Content Validation      off
   Product Level Validation     on
   Max Errors                   100000
   Registered Contexts File     /home/niessner/Projects/PDS/validate/src/main/resources/util/registered_context_products.json



Product Level Validation Results

  PASS: file:/home/niessner/Projects/PDS/validate/src/test/resources/github1210/test_case_Valid_20241024.xml
        1 product validation(s) completed

Summary:

  1 product(s)
  0 error(s)
  0 warning(s)

  Product Validation Summary:
    1          product(s) passed
    0          product(s) failed
    0          product(s) skipped
    1          product(s) total

  Referential Integrity Check Summary:
    0          check(s) passed
    0          check(s) failed
    0          check(s) skipped
    0          check(s) total


End of Report
Completed execution in 6943 ms
```

**Comment by jordanpadams** [comment]
@al-niessner yes. this actually came up in our automated testing with the PDS4 Information Model, which pulls the latest source.

**Comment by jordanpadams** [comment]
@al-niessner I think they are interchangeable? `jp2` can be a file extension and can be an encoding type. but the file extension and encoding type do not need to match.

**Comment by rsjoyner** [comment]
In SR, From Table 6C-1 Reserved File Name Extensions:

jp2 == JPEG 2000 formatted file.

**Comment by al-niessner** [comment]
@jordanpadams 

Clarify what you want. JP2 is a subset of or covered by J2C, which is JPEG 2000,  according to Mr Google and its AI assistant. What does PDS want: Do you want an encoding type of JP2 that accepts what file extensions? Do you want overlap with J2C encoding file extensions? If so, what overlap?

**Comment by jordanpadams** [comment]
JP2 should support all J2C file extensions.

**Comment by jordanpadams** [comment]
@al-niessner ☝

**Comment by rsjoyner** [comment]
Here is the list of updated encoding_standard_id's where 'JP2' was added:
```
  <sch:pattern>
    <sch:rule context="pds:Encoded_Image/pds:encoding_standard_id">
      <sch:assert test=". = ('GIF', 'J2C', 'JP2', 'JPEG', 'PDF', 'PDF/A', 'PNG', 'TIFF')">
        <title>pds:Encoded_Image/pds:encoding_standard_id/pds:encoding_standard_id</title>
        The attribute pds:Encoded_Image/pds:encoding_standard_id must be equal to one of the following values 'GIF', 'J2C', 'JP2', 'JPEG', 'PDF', 'PDF/A', 'PNG', 'TIFF'.</sch:assert>
    </sch:rule>
  </sch:pattern>
```

In SR, From Table 6C-1 Reserved File Name Extensions:
    -- jp2 == JPEG 2000 formatted file.

Note that I think the only allowable file extension for JP2 is 'filename.jp2'.

In DPH,  Table 4-3. Values for Encoded_Image .encoding_standard_id: 

GIF
J2C
JP2
JPEG
PDF
PDF/A
PNG
TIFF

**Comment by al-niessner** [comment]
@jordanpadams 

I added JP2 encoding and get this error:
```
  FAIL: file:/home/niessner/Projects/PDS/validate/src/test/resources/github1210/test_case_Valid_20241024.xml
      ERROR  [error.label.schematron]   line 161, 29: The attribute pds:Encoded_Image/pds:encoding_standard_id must be equal to one of the following values 'GIF', 'J2C', 'JPEG', 'PDF', 'PDF/A', 'PNG', 'TIFF'.
        1 product validation(s) completed
```

I do not think the failure is what you think it is or this test is broken because it points to wrong schematron. I still think your XML should be a J2C encoding.

**Comment by rsjoyner** [comment]
> [@jordanpadams](https://github.com/jordanpadams)
> 
> I added JP2 encoding and get this error:
> 
> ```
>   FAIL: file:/home/niessner/Projects/PDS/validate/src/test/resources/github1210/test_case_Valid_20241024.xml
>       ERROR  [error.label.schematron]   line 161, 29: The attribute pds:Encoded_Image/pds:encoding_standard_id must be equal to one of the following values 'GIF', 'J2C', 'JPEG', 'PDF', 'PDF/A', 'PNG', 'TIFF'.
>         1 product validation(s) completed
> ```
> 
> I do not think the failure is what you think it is or this test is broken because it points to wrong schematron. I still think your XML should be a J2C encoding.

I don't think so.  See my comment.

```
    <Encoded_Image>
      <offset unit="byte">0</offset>
      <!-- test 'JP2' as new enumerated value -->
      <encoding_standard_id>JP2</encoding_standard_id>
    </Encoded_Image>
```

The XML has the correct value.  The SCH contains the correct set of values (per the SCR).   
Not sure why Validate thinks there is a problem.

**Comment by jordanpadams** [comment]
@al-niessner @rsjoyner right, I think this checks out? @rsjoyner so we should expect 1 error there? The point here is we do not an internal error to be thrown, which is great.

**Comment by al-niessner** [comment]
@rsjoyner 

It loads the schematron from the web if it can. Is this an upgrade that is not quite public yet and so I am not loading it? If not, then, when validate used the schematron, it was clearly told that JP2 was not part of the know enumeration. It then lists all known values and JP2 is not there.

**Comment by al-niessner** [comment]
@rsjoyner 

Here is the schematron in the XML:
```
<?xml-model href="https://pds.nasa.gov/pds4/pds/v1/PDS4_PDS_1N00.sch" schematypens="http://purl.oclc.org/dsdl/schematron"?>
```

**Comment by rsjoyner** [comment]
>  so we should expect 1 error there? 

No no no.   Oxygen validates the Valid XML file (i.e., green) when validating against the 1.O.0.0 schemas.   Validate has some hidden bug / issue that is generating the Error.

[test_case_Valid_20241024.zip](https://github.com/user-attachments/files/19675328/test_case_Valid_20241024.zip)

**Comment by al-niessner** [comment]
Ah, the test data I was given was N as you can see from the headers:
```
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="https://pds.nasa.gov/pds4/pds/v1/PDS4_PDS_1N00.sch" schematypens="http://purl.oclc.org/dsdl/schematron"?>

<Product_Observational xmlns="http://pds.nasa.gov/pds4/pds/v1" 
    xmlns:pds="http://pds.nasa.gov/pds4/pds/v1" 
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" 
    xsi:schemaLocation="http://pds.nasa.gov/pds4/pds/v1 https://pds.nasa.gov/pds4/pds/v1/PDS4_PDS_1N00.xsd">
```

Explains some of it.

**Comment by al-niessner** [comment]
@rsjoyner 

Your test set fails too because changes are not public enough (aka work in progress):
```
      WARNING  [warning.label.schema]   The schema version(s) [/PDS4/PDS/V1/PDS4_PDS_1L00] does/do not match the schematron version(s) [].
      ERROR  [error.label.schematron]   Schematron 'file:/home/niessner/Projects/PDS/validate/src/test/resources/github1210/PDS4_PDS_1N00_ccb48_edited_20241022.sch' error: Cannot read schematron from URL file:/home/niessner/Projects/PDS/validate/src/test/resources/github1210/PDS4_PDS_1N00_ccb48_edited_20241022.sch
    Begin Schematron: file:/home/niessner/Projects/PDS/validate/src/test/resources/github1210/PDS4_PDS_1N00_ccb48_edited_20241022.sch
      FATAL_ERROR  [error.label.schematron]   Cannot read schematron from URL file:/home/niessner/Projects/PDS/validate/src/test/resources/github1210/PDS4_PDS_1N00_ccb48_edited_20241022.sch
    End Schematron: file:/home/niessner/Projects/PDS/validate/src/test/resources/github1210/PDS4_PDS_1N00_ccb48_edited_20241022.sch
        1 product validation(s) completed
```

**Comment by rsjoyner** [comment]
Here are the actual 1.O schemas that I have been testing against:

[CCB-48_add_JP2_to_encoding_standard_id.zip](https://github.com/user-attachments/files/19675411/CCB-48_add_JP2_to_encoding_standard_id.zip)

**Comment by jordanpadams** [comment]
@al-niessner for this particular test, we can just leave it as 1.23.0.0, and just verify it does not throw an internal error. the other tests can happen in PDS4 IM.

**Comment by al-niessner** [comment]
@jordanpadams 

Let me know what you want to do:
1. I can commit code without unit test (it works but for lack of newest schema/schematron). Your model will fail because it too is pointing at N rather than O. Wait for schema/schematron to hit the shelves, update test cases when they arrive if we care.
2. Do nothing for now. Wait for schema/schematron to hit the shelves, update test cases when they arrive. Make code changes then and go through the normal cycle.
3. Both of us use the WIP schema/schematron modifying both test cases to load local files.

They are in my personal order of choice.

**Comment by jordanpadams** [comment]
@al-niessner 1 is fine.
